### PR TITLE
avoid prime rundway skew

### PR DESCRIPTION
**Pull request by calvinmetcalf**
basically the way were were previously doing it, randomply pick number, then increment until we get a prime, would skew towards certain primes, those with a longer runway ahead of them, so say we are looking for a prime 3 bit prime, of the 8 posiblities

- 0. increment twice -> 2
- 1. increment once -> 2
- 2. prime -> 2
- 3. prime -> 3
- 4. increment once -> 5
- 5. prime -> 5
- 6. increment once -> 7
- 7. prime- > 7

in other words

- 2. 3/8
- 3. 1/8
- 5. 2/8
- 7 2/8

so 2 would be over represented while 3 would be under represented

### CODE REVIEW

**Comment by dcousens** [comment]
Was the original implementation based on any formal algorithm?
I'm not inclined to disagree with your analysis and my rudimentary look over of the code involved, but I haven't looked into it enough to be sure.

**Comment by calvinmetcalf** [comment]
I based it off how openssl works I want to say, but loosely

**Comment by calvinmetcalf** [comment]
@dcousens ping

**Comment by dcousens** [comment]
So it seems now we're just randomly iterating through?
Thoughts on [this](https://crypto.stackexchange.com/questions/1812/how-to-better-generate-large-primes-sieving-and-then-random-picking-or-random-p)?

File: lib/generatePrime.js [hunk 1]
```diff
@@ -71,62 +71,35 @@ function findPrime(bits, gen) {
...
   }
   gen = new BN(gen);
-  var runs, comp;
-  function generateRandom(bits) {
-    runs = -1;
-    var out = new BN(randomBytes(Math.ceil(bits / 8)));
-    while (out.bitLength() > bits) {
-      out.ishrn(1);
+
+  var num, n2;
```
Comment (dcousens) [inline lib/generatePrime.js]: These need not be outside the loop

Comment (calvinmetcalf) [inline lib/generatePrime.js]: they don't need to be, but it's pretty helpful when it comes to loops

Comment (dcousens) [inline lib/generatePrime.js]: How so?

**Comment by calvinmetcalf** [comment]
right we are just randomly iterating, there might be a speedup via using that method but isn't that just a time memory trade off in the end?

**Comment by dcousens** [comment]
@calvinmetcalf AFAIK yes,  but the amount of memory might negligible compared to the time.
